### PR TITLE
Bullet margin fix and Aabb query

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context
Bullet collision shapes have a [default margin on 0.04](https://pybullet.org/Bullet/BulletFull/btCollisionMargin_8h.html#ac3542bba5bfc2fcc6bbb2c8c991b947c). 

We allow the user to assign a single collision margin value for each rigid body. We assign this to the root of the btCompoundShape object which holds all sub-components of the collision shape. We have not been explicitly overriding the default margins for these sub-objects. This caused a difference between joined and component hierarchy objects in the actual collision margins of the whole, as sub-component margins are accumulated additively (i.e. default margins (0.04) on sub-components would be added to the user defined margin).

This PR fixes said issue.

Sub-thesis: 
 - Fix `PhysicsTest.cpp` build without `--bullet` issue. (identified by @matthewjmay ). 
 - Added here to avoid git churn because this PR contains the first functionality which requires the linking and better fix on master would be to remove the broken include.

## How Has This Been Tested

This PR introduces a test to be sure that different collision shape modes (scene, joined, hierarchal) will result in the correct margin. To do so, we add functionality for querying collision Aabb from Bullet and check this against ground truth for a known test asset composed of multiple parts. This test will be extended to encompass additional collision modes such as bounding boxes when added.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by aclegg3** [comment]
Ready for review.

File: src/esp/physics/bullet/BulletRigidObject.cpp [hunk 1]
```diff
@@ -177,6 +177,8 @@ bool BulletRigidObject::initializeObject(
   if (joinCollisionMeshes) {
     btTransform t;
     t.setIdentity();
+    bObjectConvexShapes_.back()->setMargin(0.0);
```
Comment (msbaines) [inline src/esp/physics/bullet/BulletRigidObject.cpp]: I haven't dug too deep in the physics code yet but why is this here and not in constructBulletCompoundFromMeshes

File: src/tests/PhysicsTest.cpp [hunk 2]
```diff
@@ -131,3 +134,62 @@ TEST_F(PhysicsManagerTest, JoinCompound) {
...
+
+    std::pair<Magnum::Vector3, Magnum::Vector3> AabbScene =
+        bPhysManager->getSceneCollisionShapeAabb();
+
+    std::pair<Magnum::Vector3, Magnum::Vector3> AabbOb0 =
+        bPhysManager->getCollisionShapeAabb(objectId0);
+    std::pair<Magnum::Vector3, Magnum::Vector3> AabbOb1 =
+        bPhysManager->getCollisionShapeAabb(objectId1);
+
+    std::pair<Magnum::Vector3, Magnum::Vector3> objectGroundTruth =
```
Comment (msbaines) [inline src/tests/PhysicsTest.cpp]: Is this test testing all the places where new code was added.

File: src/esp/physics/bullet/BulletPhysicsManager.cpp [hunk 3]
```diff
@@ -195,6 +195,18 @@ double BulletPhysicsManager::getSceneRestitutionCoefficient() const {
       ->getRestitutionCoefficient();
 }
 
+const std::pair<Magnum::Vector3, Magnum::Vector3>
```
[the remark this answers is not in the document]
Comment (aclegg3) [reply src/esp/physics/bullet/BulletPhysicsManager.cpp]: Could be. Also could be used to construct a `Magnum::Range3D`. Currently, this just matches the return type format of the Bullet query function. Since it is not exposed to the base `PhysicsManager` API, I left it that way.

File: src/esp/physics/bullet/BulletRigidObject.cpp (hunk 1 above)
Comment (aclegg3) [reply src/esp/physics/bullet/BulletRigidObject.cpp]: `constructBulletCompoundFromMeshes` is a recursive function. 

With the `join` flag, as in this case, it is building up a single convex shape from all components. This will not be complete until after the recursion is done. 

Without the `join` flag, each level of recursion builds a new object, so it makes sense to set margin upon construction in that case.

File: src/tests/PhysicsTest.cpp (hunk 2 above)
Comment (aclegg3) [reply src/tests/PhysicsTest.cpp]: Yes, there are 3 cases currently:
1. Scenes (concave mesh collision objects)
2. Joined objects (a single convex)
3. Hierarchal objects (a compound of convex components)

File: src/esp/physics/bullet/BulletRigidObject.cpp (hunk 1 above)
Comment (msbaines) [reply src/esp/physics/bullet/BulletRigidObject.cpp]: The constructBulletCompoundFromMeshes code might be easier to read with two functions. A top-level function and a recursive helper. It feels like functionality that should be part of constructBulletCompoundFromMeshes is part of initializeObject.

File: src/esp/physics/bullet/BulletPhysicsManager.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (msbaines) [reply src/esp/physics/bullet/BulletPhysicsManager.cpp]: Structs are sometimes preferred to pair when the type is used in a large scope (especially if it is used in a header file). pairs are preferred for smaller scopes (e.g. a single function) and can result in hard to read code when used in a large scope. It is somewhat analogous to the use of the variable name "i" in a for loop.

File: src/tests/PhysicsTest.cpp [hunk 4]
```diff
@@ -131,3 +134,62 @@ TEST_F(PhysicsManagerTest, JoinCompound) {
...
+        std::pair<Magnum::Vector3, Magnum::Vector3>({-1.1, -1.1, -1.1},
+                                                    {1.1, 1.1, 1.1});
+    std::pair<Magnum::Vector3, Magnum::Vector3> sceneGroundTruth =
+        std::pair<Magnum::Vector3, Magnum::Vector3>({-1.0, -1.0, -1.0},
+                                                    {1.0, 1.0, 1.0});
+
+    // Cr::Utility::Debug() << "aabb scene: " << AabbScene;
+    // Cr::Utility::Debug() << "aabb ob0: " << AabbOb0;
+    // Cr::Utility::Debug() << "aabb ob1: " << AabbOb1;
```
Comment (msbaines) [inline src/tests/PhysicsTest.cpp]: These should be uncommented or removed.

File: src/esp/physics/bullet/BulletRigidObject.cpp (hunk 1 above)
Comment (aclegg3) [reply src/esp/physics/bullet/BulletRigidObject.cpp]: Fair point. 👍 
A refactor like this should probably be a separate PR after this bug fix is merged.
